internal/kafka/internal/services/cloud_providers_test.go

```diff
@@ -485,10 +498,51 @@ func Test_ListCloudProviderRegions(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "An error is returned when the cloud provider cannot be obtained",
+			fields: fields{
+				connectionFactory: db.NewMockConnectionFactory(nil),
+				providerFactory: &clusters.ProviderFactoryMock{
+					GetProviderFunc: func(providerType api.ClusterProviderType) (clusters.Provider, error) {
+						return nil, errors.New("failed to get provider implementation")
+					},
+				},
+			},
+			wantErr: true,
+			setupFn: func() {},
```

Comment:
[optional] this way you are creating a different function everywhere. Since the empty function is pretty common in these tests, I'd suggest to create a global variable:
```go
emptyFunc := func(){}
```
and then change these occurrences with:
```go
setupFn: emptyFunc
```